src/inference/src/compilation_context.cpp

```diff
@@ -41,9 +41,83 @@ static int32_t as_int32_t(T v) {
     return static_cast<int32_t>(v);
 }
 
-//////////////////////////////////////////////////
+}  // namespace ov
+
+namespace {
+
+uint64_t compute_model_hash(const std::shared_ptr<const ov::Model>& model, const ov::AnyMap& compileOptions) {
+    OPENVINO_ASSERT(model);
+
+    uint64_t seed = 0;
+    // 1. Calculate hash on function
+    ov::pass::Manager m;
+    m.register_pass<ngraph::pass::FixRtInfo>();
+    m.register_pass<ov::pass::Hash>(seed);
+    m.run_passes(std::const_pointer_cast<ov::Model>(model));
+
+    // 2. Compute hash on serialized data and options
+    for (const auto& kvp : compileOptions) {
+        seed = ov::hash_combine(seed, kvp.first + kvp.second.as<std::string>());
+    }
+
+    // 3. Add runtime information which may not be serialized
+    for (const auto& op : model->get_ordered_ops()) {
+        const auto& rt = op->get_rt_info();
+        for (const auto& rtMapData : rt) {
+            seed = ov::hash_combine(seed, rtMapData.first);
+            std::stringstream strm;
+            rtMapData.second.print(strm);
+            seed = ov::hash_combine(seed, strm.str());
+        }
+    }
+
+    // 4. Legacy part if CNNNetwork is used with new Plugin API
+    for (auto&& input : model->inputs()) {
```

Comment:
do we have cases when caching works with ov::Model containing old ie_legacy_preproc ? 
I supposed, that caching in case of OLD API works only with CNNNetwork structures.

Reply:
Old API works with new ov::CoreImpl. So yes we need to have this code.

Reply:
as I see, Old API LoadNetwork is fully duplicated, so we should not have mix when LoadNetwork goes to ov::Model with old preprocessing, is not it?